src/engine/uct/node/mod.rs

```diff
@@ -146,21 +146,38 @@ impl Node {
                 .iter()
                 .map(|&m| Node::new(m, self.config))
                 .collect();
+            if self.children.len() <= (game.size() * game.size() / 10) as usize {
+                let player = game.last_move().color().opposite();
+                if game.winner() == player || !self.config.uct.playout_aftermath { //don't pass if we're losing on the board
+                    self.children.push(Node::new(Pass(player), self.config));
+                }
+            }
+            
             self.descendants = self.children.len();
         }
  }
 
     pub fn expand(&mut self, board: &Board) -> bool {
         let not_terminal = !board.is_game_over();
-        let mut children = vec![];
         if not_terminal && self.plays >= self.config.uct.expand_after {
-            children = board.legal_moves_without_eyes()
+            let mut children = board.legal_moves_without_eyes()
                 .iter()
                 .map(|m| self.new_leaf(board, m))
                 .collect();
+                
+            self.priors(&mut children, board);
+            self.children = children;
+            
+            if self.children.len() <= (board.size() * board.size() / 10) as usize {
+                let player = board.next_player();
+                if board.winner() == player { //don't pass if we're losing on the board
```

Comment:
next_player is the player we're considering the moves for

so if we're looking at the White player's moves, we will only pass as White if White is the winner on the board

if White is not winning on the board, we can't pass because the enemy can immediately pass afterward on CGOS and win the game (due to stones being on the board that are not alive but can't be removed due to CGOS rules)